lxd/daemon.go

```diff
@@ -157,36 +157,38 @@ type Command struct {
 }
 
 // Check whether the request comes from a trusted client.
-func (d *Daemon) checkTrustedClient(r *http.Request) error {
+func (d *Daemon) checkTrustedClient(r *http.Request) (error, string) {
```

Comment:
It feels it'd be a bit better to leave the signature of this API unchanged, and add a separate API, e.g.:

```go
func (d *Daemon) authenticateRequest(r *http.Request) (string, error)
```

which would have the same implementation of the ```checkTrustedClient``` in this branch, except that the order of the return values would be the regular one as opposed to the current ```(error, string)```. Then you would implement ```checkTrustedClient``` as:

```go
func (d *Daemon) checkTrustedClient(r *http.Request) error {
    _, err := d.authenticateRequest(r)
   return err
}
```

While you are on it, since the error object returned by ```checkTrustedClient``` is not used in any way, and what call sites are really interested in is a yes/no answer, I'd drop the ```checkTrustedClient``` method in favor of a ```isClientTrusted``` method:

```go
func (d *Daemon) isClientTrusted(r *http.Request) bool {
    _, err := d.authenticateRequest(r)
   return err == nil
}
```